### PR TITLE
Refs [TASK][371013] Data Teams Screen

### PR DESCRIPTION
**Pull request by zGararz**
## Related Tickets
- [#Task 371013: Data Teams Screen](https://edu-redmine.sun-asterisk.vn/issues/371013)
## WHAT
- 
## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://github.com/framgia/coding-standards/tree/master/eng/android |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Conventions | Kotlin coding conventions | https://kotlinlang.org/docs/coding-conventions.html |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/sunasterisk/loltournaments/ui/teams/TeamsFragment.kt [hunk 1]
```diff
@@ -1,7 +1,33 @@
...
 import com.sunasterisk.loltournaments.R
+import com.sunasterisk.loltournaments.base.BaseFragment
+import com.sunasterisk.loltournaments.data.model.remote.Team
+import com.sunasterisk.loltournaments.databinding.FragmentTeamsBinding
+import com.sunasterisk.loltournaments.ui.adapter.TeamAdapter
+import org.koin.androidx.viewmodel.ext.android.viewModel
 
-class TeamsFragment : Fragment(R.layout.fragment_more) {
+class TeamsFragment : BaseFragment<FragmentTeamsBinding>() {
+    override val layoutResource = R.layout.fragment_teams
```
Comment (phucnv-2558) [inline app/src/main/java/com/sunasterisk/loltournaments/ui/teams/TeamsFragment.kt]:
```suggestion
    override val layoutResource 
        get() = R.layout.fragment_teams
```

File: app/src/main/java/com/sunasterisk/loltournaments/ui/teams/TeamsViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+            if (currentPage.value != FIRST_PAGE) {
+                teams.value?.let { currentTeams.addAll(it) }
+            }
+
+            currentTeams.addAll(teamsLoaded)
+            _teams.postValue(currentTeams)
+
+            if (teamsLoaded.isNullOrEmpty()) _isLastPage.postValue(true)
+
```
Comment (phucnv-2558) [inline app/src/main/java/com/sunasterisk/loltournaments/ui/teams/TeamsViewModel.kt]: space u

File: app/src/main/java/com/sunasterisk/loltournaments/ui/teams/TeamsFragment.kt (hunk 1 above)
Comment (zGararz) [reply app/src/main/java/com/sunasterisk/loltournaments/ui/teams/TeamsFragment.kt]: t thấy nó có khác gì đâu nhỉ ?

Comment (huytq-2422) [reply app/src/main/java/com/sunasterisk/loltournaments/ui/teams/TeamsFragment.kt]: khác đấy nhé, em xem kotlin byte code nhé

**Review by huytq-2422 — APPROVED** [review]

**Pull request merged**